hail/src/main/scala/is/hail/expr/types/physical/PSubsetStruct.scala

```diff
@@ -0,0 +1,214 @@
+
+  override def setFieldPresent(structAddress: Long, fieldIdx: Int): Unit = ???
+
+  override def setFieldPresent(structAddress: Code[Long], fieldIdx: Int): Code[Unit] = ???
+
+  override def appendKey(key: String, sig: PType): PStruct = ???
+
+  def insertFields(fieldsToInsert: TraversableOnce[(String, PType)]): PSubsetStruct = ???
+
+  override def initialize(structAddress: Long, setMissing: Boolean): Unit = ???
+
+  override def stagedInitialize(structAddress: Code[Long], setMissing: Boolean): Code[Unit] = ???
+
+  override def allocate(region: Region): Long = ???
+
+  override def allocate(region: Code[Region]): Code[Long] = ???
+
+  override def setRequired(required: Boolean): PType = ???
+
+  override def copyFromType(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Long] = ???
+
+  override def copyFromTypeAndStackValue(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, stackValue: Code[_], deepCopy: Boolean): Code[_] = ???
+
+  override protected def _copyFromAddress(region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Long = ???
+
+  override def constructAtAddress(mb: EmitMethodBuilder[_], addr: Code[Long], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Unit] = ???
+
+  override def constructAtAddress(addr: Long, region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Unit = ???
+}
+
+object PSubsetStructSettable {
+  def apply(cb: EmitCodeBuilder, pt: PStruct, name: String, sb: SettableBuilder): PSubsetStructSettable = {
+    new PSubsetStructSettable(pt, sb.newSettable(name))
+  }
+}
+
+class PSubsetStructSettable(val pt: PStruct, val a: Settable[Long]) extends PBaseStructValue with PSettable {
+  def get: PSubsetStructCode = new PSubsetStructCode(pt, a)
+
+  // Again, is fieldIdx here the backing PStruct?
+  def loadField(cb: EmitCodeBuilder, fieldIdx: Int): IEmitCode = {
+    IEmitCode(cb,
+      pt.isFieldMissing(a, fieldIdx),
+      pt.fields(fieldIdx).typ.load(pt.fieldOffset(a, fieldIdx)))
+  }
+
+  def store(pv: PCode): Code[Unit] = {
+    a := pv.asInstanceOf[PSubsetStructCode].a
+  }
+}
+
+// TODO: We don't allow construction, so what do we do here for store?
+class PSubsetStructCode(val pt: PStruct, val a: Code[Long]) extends PBaseStructCode {
+  def code: Code[_] = a
+
+  def codeTuple(): IndexedSeq[Code[_]] = FastIndexedSeq(a)
+
+  def memoize(cb: EmitCodeBuilder, name: String, sb: SettableBuilder): PBaseStructValue = {
+    val s = PSubsetStructSettable(cb, pt, name, sb)
+    cb.assign(s, this)
+    s
+  }
+
+  def memoize(cb: EmitCodeBuilder, name: String): PBaseStructValue = memoize(cb, name, cb.localBuilder)
+
+  def memoizeField(cb: EmitCodeBuilder, name: String): PBaseStructValue = memoize(cb, name, cb.fieldBuilder)
+
+  def store(mb: EmitMethodBuilder[_], r: Value[Region], dst: Code[Long]): Code[Unit] =
```

Comment:
and in line with the above comment, should this remain unimplemented.

Reply:
hmm, good question. Need to think about that.

Reply:
After some more thought, probably the right thing to do for now is call `store` on the parent PStructCode.